src/InstallTaskManager.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Drupal\stanford_profile;
+
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Plugin\DefaultPluginManager;
+
+/**
+ * Install task plugin manager.
+ *
+ * @package Drupal\stanford_profile
+ * @codeCoverageIgnore
+ *   We can't test a service in profile due to some limitations of the Kernel.
+ */
+class InstallTaskManager extends DefaultPluginManager {
+
+  /**
+   * Array of completed plugin ids.
+   *
+   * @var array
+   */
+  protected $completedTasks = [];
+
+  /**
+   * Constructs a ArchiverManager object.
+   *
+   * @param \Traversable $namespaces
+   *   An object that implements \Traversable which contains the root paths
+   *   keyed by the corresponding namespace to look for plugin implementations.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache_backend
+   *   Cache backend instance to use.
+   * @param \Drupal\Core\Extension\ModuleHandlerInterface $module_handler
+   *   The module handler to invoke the alter hook with.
+   */
+  public function __construct(\Traversable $namespaces, CacheBackendInterface $cache_backend, ModuleHandlerInterface $module_handler) {
+    parent::__construct(
+      'Plugin/InstallTask',
+      $namespaces,
+      $module_handler,
+      'Drupal\stanford_profile\InstallTaskInterface',
+      'Drupal\stanford_profile\Annotation\InstallTask'
+    );
+    $this->alterInfo('install_task_plugins');
+    $this->setCacheBackend($cache_backend, 'install_task_plugins');
+  }
+
+  /**
+   * Run all install task plugins.
+   *
+   * @param array $install_state
+   *   Current install state.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function runTasks(array &$install_state) {
+    foreach ($this->getDefinitions() as $definition) {
+      $this->runTask($definition, $install_state);
+    }
+  }
+
+  /**
+   * Run the given task after any dependencies.
+   *
+   * @param array $task_definition
+   *   Plugin definition.
+   * @param array $install_state
+   *   Current install state.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  protected function runTask(array $task_definition, array &$install_state) {
+    foreach ($task_definition['dependencies'] as $dependency) {
```

Comment:
This could lead us into a circular dependency loop and never end.

Reply:
like if a task declares itself?

Reply:
Task1 depends on Task2 depends on Task3 depends on Task1
Rinse
Repeat

Reply:
https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Graph%21Graph.php/class/Graph/8.7.x

The update hook dependency resolution uses this Graph API. I'm not sure how it works or whether it is useful but here is the info anyhow.

https://api.drupal.org/api/drupal/core%21includes%21update.inc/function/update_resolve_dependencies/8.7.x